tests/test_cases/test_cocotb/test_cocotb.py

```diff
@@ -1316,4 +1316,14 @@ def test_bad_attr(dut):
         assert False, "Expected AttributeError"
 
 
+@cocotb.test(skip=cocotb.SIM_NAME.lower().startswith("icarus"))
+async def test_string_handle_takes_bytes(dut):
+    test_string = "bytes"
+    dut.string_input_port.value = test_string.encode("ascii")
+    await cocotb.triggers.Timer(10, 'ns')
+    val = dut.string_input_port.value
+    assert isinstance(val, bytes)
+    assert val.decode("ascii") == test_string
```

Comment:
Perhaps simpler to avoid an encoding dance here:
```suggestion
    dut.string_input_port.value = b"bytes"
    await cocotb.triggers.Timer(10, 'ns')
    val = dut.string_input_port.value
    assert isinstance(val, bytes)
    assert val == b"bytes"
```